### ISSUE TITLE
All Module Options are broken

### ISSUE THREAD

**Issue by brianteeman** [joomla/joomla-cms#38690]
### Steps to reproduce the issue
Create a module and put it in any of the foillowing positions in the cassiopeia template
- topbar
- below-top
- menu
- search
- banner
- breadcrumbs
- footer

Set the title to Show
Set any of the fields in the Advanced tab

### Expected result
Title is displayed
Value of the fields are used

### Actual result
No title
Values are ignored

### Additional comments
The reason for this is that the template uses the chrome/style of "none" for all these positions. The chrome doesnt support title or any of the advanced fields.

This is just rediculous. What is the point in having all these options if they cant be used. I am sick and tired of answering questions related to this on the forums and elsewhere.

I'm an advanced user so I know I can change the Style in the advanced tab to a more sensible chrome/style such as html5 but for everyone else joomla is **broken and doesnt work**

Its not even as if I can get a clue about the style that is being used for a position without digging into the php of the template because it just says "inherited".

#### Option 1 
change the template to use style=html5 instead of style=none

#### Option 2
change the chrome none.php to support all the options

**Comment by brianteeman** [comment]
cc @crystalenka @coolcat-creations @angieradtke JXT members

**Comment by crystalenka** [comment]
This should be a relatively quick fix, but may break some templates/overrides/sites because the output is different (which is part of the problem). If someone styled a site based on the style=none modules, it may not work if the default style changes.

@bembelimen I can open a PR for this, but would it be a breaking change? If so, how big of a breaking change? Seems silly to wait for J5, but it is a frustration. Would it be possible to set the default for new installations without breaking existing sites, or something?

**Comment by brianteeman** [comment]
One of the problems is that when you create a new module show title is set to yes
So if the template is changed from "none" to "html5" then the titles will magically appear on all sites

**Comment by coolcat-creations** [comment]
if I set it to html5 another problem appears: the header tag is nested and twice:
![grafik](https://user-images.githubusercontent.com/828371/188434776-cc14e175-847b-43ca-8de4-940f2d5bc708.png)

**Comment by coolcat-creations** [comment]
I think it would be nice to have "cards" for the different Chrome Styles and display them with a preview of what the user can expect. Something like small wireframes. I could draw them if there is interest in that. But we would need a feature to attach a svg icon to preview the different styles.

**Comment by brianteeman** [comment]
My only concerns right now is
1. module options dont work and almost no one knows why

**Comment by coolcat-creations** [comment]
> My only concerns right now is
> 
>     1. module options dont work and almost no one knows why

I agree. Also for example the Bootstrap size does not work for anything as far as I am concerned. Correct me if I am wrong.

**Comment by coolcat-creations** [comment]
Can we check if the chrome is using title or setting at all and if not display a warnung below the setting?

**Comment by brianteeman** [comment]
> Can we check if the chrome is using title or setting at all and if not display a warnung below the setting?

no. because we dont actually know what chrome is being used. the module only knows it is using "inherited"

**Comment by coolcat-creations** [comment]
But inherited from what is a big question. From the template? then it has to check the template. But I fear thats quite impossible...

**Comment by Fedik** [comment]
Well, It not something new, it expected behavior. Probably since we have "chrome" style. Most of "joomlers" aware about it.
Yes, I understand it looks unintuitive for beginners, and may confuse.

**Comment by crystalenka** [comment]
> Well, It not something new, it expected behavior. Probably since we have "chrome" style. Most of "joomlers" aware about it.
> Yes, I understand it looks unintuitive for beginners, and may confuse.

It is not expected behavior that the default chrome will not actually respect any of the module settings. Also, in Joomla 3, the default was html (I believe) so it worked as expected. 

It may not be new, but that doesn't mean we can't make it better.

**Comment by crystalenka** [comment]
> > My only concerns right now is
> > ```
> > 1. module options dont work and almost no one knows why
> > ```
> 
> I agree. Also for example the Bootstrap size does not work for anything as far as I am concerned. Correct me if I am wrong.

I think you are right, as far as Cassiopeia is concerned. But if Cassiopeia will be the Bootstrap template going forward, perhaps the issue is that the template doesn't do anything with them.

**Comment by coolcat-creations** [comment]
![grafik](https://user-images.githubusercontent.com/828371/188440184-066a489f-2203-4869-9eef-e64479e0fd9a.png)

**Comment by angieradtke** [comment]
I think by default it should be, that all parameters work as desired.  So html5 should be default, also in Cassiopeia.

Maybe it needs an additional helptext   that this setting is also affected by the used  template.

I thinks icons  @coolcat-creations  are helpful , but  if the style is affected by the template, we don't know how the module will be renderd. As a developer I don't want to go  without "none". This allows me clean code.

**Comment by Fedik** [comment]
> It is not expected behavior that the default chrome will not actually respect any of the module settings. Also, in Joomla 3, the default was html (I believe) so it worked as expected.

This kind of template design flaw, it would need something different than `style="none"` by default, I not sure we can change it for Cassiopeia.
Joomla 3 also have multiple default positions set to none.

https://github.com/joomla/joomla-cms/blob/4d5e843cbcc71b40ec368179e31176f1e87d53e9/templates/protostar/index.php#L152
https://github.com/joomla/joomla-cms/blob/4d5e843cbcc71b40ec368179e31176f1e87d53e9/templates/protostar/index.php#L167
https://github.com/joomla/joomla-cms/blob/4d5e843cbcc71b40ec368179e31176f1e87d53e9/templates/protostar/index.php#L188

> It may not be new, but that doesn't mean we can't make it better.

I did not say we can't do better, I just remind that it not a new, and how whole crome thing works :wink: 
By "chrome" API design it expected behavior.

> If we know to which menu items the module is assigned to, we can find out which template styles it is using.
It can even be it's using a lot of different template styles.

And what about "render module" plugin? :wink:

**Comment by coolcat-creations** [comment]
The icons would be wireframes provided by the chromes. No specific stying but an idea for the user how the structure looks like.

**Comment by Fedik** [comment]
@coolcat-creations  yeah, maybe drop "inherit" and use "html5" not a bad idea, in result it will be more obvious what is going on.
In theory.

**Comment by coolcat-creations** [comment]
I would not drop inherit. Inherit means it uses the template style so if I have 5 Styles I can define which style is used in each of the templates instead creating 5 Modules.
But it would be good to pass the information back to the module which styles are being used. 
Normally only one - but sometimes there could be more styles in use.

**Comment by brianteeman** [comment]
> Well, It not something new, it expected behavior. Probably since we have "chrome" style. Most of "joomlers" aware about it. Yes, I understand it looks unintuitive for beginners, and may confuse.

what is new is that the chrome doesnt support the module options

**Comment by Fedik** [comment]
> what is new is that the chrome doesnt support the module options

Try the same test for joomla 3, for position `position-0`, `position-1`, `position-2`.
You will get the same result.
Position with style `none` does not customisable by design.

**Comment by chmst** [comment]
A simple solution would be: set a default layout  "html5" for some positions in the template index.php where protostar had the default value "xhtml".

**Comment by Fedik** [comment]
I think it make sense to link this issue:
- https://github.com/joomla/joomla-cms/issues/20864

**Comment by coolcat-creations** [comment]
I thought they were ported to layouts already?

**Comment by Fedik** [comment]
yes it is, but that not all :wink: 
In linked PR #23570 was discussion about allowing XML for chrome style, so each chrome would have own config fields.

oh, and that already should be possible since 4.1 #23577
